src/librustc_back/lib.rs

```diff
@@ -168,3 +168,45 @@ impl ToJson for RelroLevel {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, PartialEq, Hash, RustcEncodable, RustcDecodable)]
+pub enum VaListKind {
+    CharPtr,
+    VoidPtr,
+    AArch64Abi,
+    PowerPcAbi,
+    X86_64Abi,
```

Comment:
I wonder why this `enum` was added, instead of determining this information from the architecture, in the call ABI (formerly `src/librustc_trans/{abi,cabi_*}.rs`, now in `src/librustc_target/abi/call/*.rs`) infrastructure.

Reply:
Is there value in custom target specifications overriding it? Currently we don't support overriding most of the behavior of the C call ABI for a target.

Reply:
My main motivation for using this approach was because it more closely mirrored clang's implementation. Other than that, it makes windows a bit easier. Windows may be `x86_64`, but the `va_list` implementation is the `CharPtr` variant. That being said, I think we could use `is_windows_like` and the architecture.

Reply:
Yeah, we already special-case windows in call ABI code. Also, I'm curious what it affects, in terms of the final implementation - is it *just* required to be something that only LLVM operates on, or would `rustc_trans` also have to generate code looking at its fields etc.?

Reply:
I'm honestly not entirely sure yet. I'm still a rustc noobie, and I'm still reading through much of the `trans` code. I was hoping that `rustc_trans::Type::va_list` could be the only function that needed this enum, but the `VaList` argument would get eliminated as an argument due to the `core` implementation being a ZST.

I was also hoping that we could generate better debug info for architectures like `x86_64` and `Aarch64`, but I haven't gotten that far yet.